### DISCUSSION TITLE
basic menu in the GTK window

### DISCUSSION THREAD

**Discussion by philippe06150** [gtk-rs/gtk4-rs#1481] [Q&A]
Hi,

i do not understand the logic to show a menu in the window, I think that I understand I must use PopoverMenuBar, so I take the basic Hello World and add menu (in fn build_ui):

	let mm = PopoverMenuBar::builder()
		.build();
	mm.add_child(&button, "essai");

no error at compilation, but no menu shown on window

I understand in doc that I have to define a menuModel somewhere, I go to this page:

https://gtk-rs.org/gtk-rs-core/git/docs/gio/struct.MenuModel.html

but I do not understand what to do, there is no basic example in the documentation or https://github.com/gtk-rs/gtk4-rs/tree/master/examples.

Somebody would have a simple example, it would help ? Or may be PopoverMenuBar is not relevant (too complicated) for a basic menu and another structure is more suitable ???

Thanks for help.

**Comment by bilelmoussaoui** [comment]
What kind of menu you are trying to set up? you would usually define that in a UI file as that is much much simple then using gio::MenuModel directly.

**Reply by philippe06150** [reply]
Hi,

I would like to list the name of sensors that I can get values in real time on a network in order for an operator to choose the sensor to show

The UI file you are talking about is the Menumodel I can import with PopoverMenuBar ?

**Comment by philippe06150** [comment]
OK 

I found an example of ui file in gtk4 doc

```
<?xml version="1.0" encoding="UTF-8"?>
 <interface>
   <menu id="menubar">
     <submenu>
       <attribute name="label">File</attribute>
       <section>
         <item>
           <attribute name="label">New</attribute>
           <attribute name="action">app.new</attribute>
         </item>
         <item>
           <attribute name="label">Open</attribute>
           <attribute name="action">app.open</attribute>
         </item>
       </section>
     </submenu>
   </menu>
 </interface>
```

I put it in menu.ui in SRC folder

then in .rs i add as explain in https://gtk-rs.org/gtk4-rs/git/docs/gtk4/struct.Builder.html
	let path = Path::new("./menu.ui");
	let mm = gtk::Builder().from_file(path);

does not compile ? I give up ...

**Reply by bilelmoussaoui** [reply]
I need more context to be able to help you
